### ISSUE TITLE
Trouble with @setNLObjective from 

### ISSUE THREAD

**Issue by anriseth** [JuliaOpt/JuMP.jl#567]
Hi, 

I'm trying to do an optimization with an array of different objectives within JuMP. But I get a type error when running nonlinear solves:
```julia
using JuMP
m = Model()
@defVar(m, 0 <= x <= 1)
@defExpr(f1, -x)
@defNLExpr(f2, -x)

fs = [f1, f2]

@setObjective(m, :Min, fs[1]) #Works
solve(m)

@setNLObjective(m, :Min, f2) #Works
solve(m)

@setNLObjective(m, :Min, fs[2]) #Fails
solve(m)
```
The last line raises the error
```julia
ERROR: type: eval_f: in typeassert, expected Float64, got ParametricExpression{0}
 in eval_f at /home/asbjorn/.julia/v0.3/JuMP/src/nlp.jl:246
 in initialize at /home/asbjorn/.julia/v0.3/JuMP/src/nlp.jl:222
 in loadnonlinearproblem! at /home/asbjorn/.julia/v0.3/Ipopt/src/IpoptSolverInterface.jl:269
 in _buildInternalModel_nlp at /home/asbjorn/.julia/v0.3/JuMP/src/nlp.jl:540
 in buildInternalModel at /home/asbjorn/.julia/v0.3/JuMP/src/solvers.jl:186
 in solve at /home/asbjorn/.julia/v0.3/JuMP/src/solvers.jl:77

```

**Comment by mlubin** [comment]
This syntax isn't currently supported within the nonlinear macros. There
are too many levels of indirection to easily handle.

**Comment by mlubin** [comment]
Will be fixed by #638:

```julia
using JuMP
m = Model()
@defVar(m, 0 <= x <= 1)
@defExpr(f1, -x)
@defNLExpr(m, f2, -x)

fs = [f1, f2]

@setObjective(m, Min, fs[1]) #Works
solve(m)

@setNLObjective(m, Min, f2) #Works
solve(m)

@setNLObjective(m, Min, fs[2]) #Works
solve(m)
```
